pkg/storage/snapshot/restore.go

```diff
@@ -128,20 +123,35 @@ func (ctrl *VMRestoreController) updateVMRestore(vmRestoreIn *snapshotv1.Virtual
 	if vmRestoreDeleting(vmRestoreOut) {
 		return 0, ctrl.handleVMRestoreDeletion(vmRestoreOut, target)
 	}
+
 	if !VmRestoreProgressing(vmRestoreOut) {
 		return 0, nil
 	}
 
 	if len(vmRestoreOut.OwnerReferences) == 0 {
 		target.Own(vmRestoreOut)
+	}
+	controller.AddFinalizer(vmRestoreOut, vmRestoreFinalizer)
+
+	if !equality.Semantic.DeepEqual(vmRestoreIn.ObjectMeta, vmRestoreOut.ObjectMeta) {
+		vmRestoreOut, err = ctrl.Client.VirtualMachineRestore(vmRestoreOut.Namespace).Update(context.Background(), vmRestoreOut, metav1.UpdateOptions{})
```

Comment:
Here, you are checking only the `ObjectMeta` but updating  the entire object, what if other fields outside the metadata are changed.

Reply:
exactly, controller should never touch the spec, just the metadata. Also `Update` does not operate on the status